ChallengeTrackerApplication/ChallengeTrackerApplication-ejb/src/java/ejb/session/stateless/UserEntitySessionBean.java

```diff
@@ -23,6 +32,41 @@
     @PersistenceContext(unitName = "ChallengeTrackerApplication-ejbPU")
     private EntityManager em;
     
+    private final ValidatorFactory validatorFactory;
+    private final Validator validator;
+
+    public UserEntitySessionBean() {
+        validatorFactory = Validation.buildDefaultValidatorFactory();
+        validator = validatorFactory.getValidator();
+    }
+
+
+    @Override
+    public Long createNewUser(User newUserEntity) throws UserUsernameExistException, UnknownPersistenceException, InputDataValidationException {
```

Comment:
In general, structure the method with try, catch, finally (if any resources to be released).
You can also use methods as exception handlers in the catch block, method name should indicate, what exception it is handling. This improves code readability. These  exception handlers can be reusable too.
Variable names to be meaningful - instead of em - entityMgr

Reply:
done